### DISCUSSION TITLE
Label and value presentation om same row

### DISCUSSION THREAD

**Discussion by Jungan22** [rjsf-team/react-jsonschema-form#3876] [Q&A]
Hi, I have very large forms and to improve readability I want to present each label/value pair on the same row and without text-box formatting for the value. The use-case here is that I want to present the form content in read-only mode. Please see picture:

![image](https://github.com/rjsf-team/react-jsonschema-form/assets/10724151/b51308cc-3415-4b2b-b9a0-83bf28930dea)

I have tried to create different custom widgets and field-templates but failed to achieve the desired result. Could you please advise what kind of custom component I need to use?

Ideally, the label/value pairs should also be distributed in two columns.

**Comment by hyprdia** [comment]
You can use [JSONForms](https://jsonforms.io/) for that, it supports this out of the box.

**Reply by msueping** [reply]
I don't see any example in JSONForms that puts the label on the *same* line as the input. Did you see what the OP asked for? The input labels are in line with the input itself. Not on top.

Can you please provide a better link?

**Reply by hyprdia** [reply]
I didn't use it, but a quick look through their documentation leads me to believe that [custom renderers](https://jsonforms.io/docs/tutorial/custom-renderers) could be used in this situation.

**Comment by heath-freenome** [comment]
You may be able to override the `WrapIfAdditionalTemplate` to do what you want
